elyra/util/git.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from github import Github, GithubException
+from traitlets.config import LoggingConfigurable
+
+
+class GithubClient(LoggingConfigurable):
+
+    def __init__(self,
+                 token=None,
+                 repo=None,
+                 branch=None,
+                 ):
+
+        super().__init__()
+
+        self.client = Github(login_or_token=token)
+        self.repo = self.client.get_repo(repo)
```

Comment:
We should catch the git exception here and raise a `RuntimeError`. If we do, the error message will be a lot more useful we can avoid returning the stacktrace to the UI. For example, there are two error scenarios that can be attributed to a user error: invalid token, invalid repo and invalid server. A RuntimeError could indicate `"Error accessing repository {repo}` followed by information extracted from the original exception message. As a safety net we can always `log.err` the original exception message.